### ISSUE TITLE
VMAgent config is not generated properly then attach_metadata is set.

### ISSUE THREAD

**Issue by omegarus** [VictoriaMetrics/operator#697]
HI.
I'm using this VMServiceScrape resource to scrape data from node-exporter service:
```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMServiceScrape
metadata:
  labels:
    app.kubernetes.io/instance: victoria-metrics-k8s-stack
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: victoria-metrics-k8s-stack
    app.kubernetes.io/version: v1.91.3
    argocd.argoproj.io/instance: victoria-metrics-k8s-stack
    helm.sh/chart: victoria-metrics-k8s-stack-0.17.0
  name: victoria-metrics-k8s-stack-node-exporter
  namespace: victoria-metrics
spec:
  endpoints:
  - attach_metadata:
      node: true
    metricRelabelConfigs:
    - action: drop
      regex: /var/lib/kubelet/pods.+
      source_labels:
      - mountpoint
    port: metrics
  jobLabel: jobLabel
  selector:
    matchLabels:
      app.kubernetes.io/name: prometheus-node-exporter
```
VM operator generates following job configuration for VMagent:
```
- job_name: serviceScrape/victoria-metrics/victoria-metrics-k8s-stack-node-exporter/0
  honor_labels: false
  kubernetes_sd_configs:
  - role: endpoints
    attach_metadata:
      node: "true"
    namespaces:
      names:
      - victoria-metrics
  relabel_configs:
  ....
```
Looks like operator converting boolean true to string.
VMAgent fails to read this config:
```
2023-07-21T23:08:20.718Z	info	VictoriaMetrics/lib/promscrape/scraper.go:114	reading Prometheus configs from "/etc/vmagent/config_out/vmagent.env.yaml"
2023-07-21T23:08:20.723Z	fatal	VictoriaMetrics/lib/promscrape/scraper.go:117	cannot read "/etc/vmagent/config_out/vmagent.env.yaml": cannot parse Prometheus config from "/etc/vmagent/config_out/vmagent.env.yaml": cannot unmarshal data: yaml: unmarshal errors:
  line 368: cannot unmarshal !!str `true` into bool; pass -promscrape.config.strictParse=false command-line flag for ignoring unknown fields in yaml config
```

**Comment by Haleygo** [comment]
Thanks for report!

**Comment by Amper** [comment]
Fixed at [v0.36.0](https://github.com/VictoriaMetrics/operator/releases/tag/v0.36.0) release
